<!DOCTYPE html>
<html>
<head>
  <meta charset="utf-8">
  <meta name="generator" content="pandoc">
  <meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">
  <title></title>
  <style type="text/css">code{white-space: pre;}</style>
  <!--[if lt IE 9]>
    <script src="http://html5shim.googlecode.com/svn/trunk/html5.js"></script>
  <![endif]-->
  <style type="text/css">
table.sourceCode, tr.sourceCode, td.lineNumbers, td.sourceCode {
  margin: 0; padding: 0; vertical-align: baseline; border: none; }
table.sourceCode { width: 100%; line-height: 100%; }
td.lineNumbers { text-align: right; padding-right: 4px; padding-left: 4px; color: #aaaaaa; border-right: 1px solid #aaaaaa; }
td.sourceCode { padding-left: 5px; }
code > span.kw { color: #007020; font-weight: bold; }
code > span.dt { color: #902000; }
code > span.dv { color: #40a070; }
code > span.bn { color: #40a070; }
code > span.fl { color: #40a070; }
code > span.ch { color: #4070a0; }
code > span.st { color: #4070a0; }
code > span.co { color: #60a0b0; font-style: italic; }
code > span.ot { color: #007020; }
code > span.al { color: #ff0000; font-weight: bold; }
code > span.fu { color: #06287e; }
code > span.er { color: #ff0000; font-weight: bold; }
  </style>
</head>
<body>
<p>Data and scripts for this Tutorial and Lectures are available at <a href="https://github.com/hyphaltip/CSHL_NGS">https://github.com/hyphaltip/CSHL_NGS</a>.</p>
<p>See the answers at <a href="/CSHL_NGS/lecture/Tutorial_steps.table.html">Tutorial answers</a> and the raw version of this file is in <a href="/CSHL_NGS/lecture/Tutorial_steps.table.md">Tutorial_steps.table.md</a></p>
<h1 id="preparation-steps">Preparation Steps</h1>
<ol type="1">
<li>Checkout my github repo for this course.</li>
</ol>
<p>Using Git on the command line (or install Git for Mac)</p>
<pre class="sourceCode perl"><code class="sourceCode perl">$ git clone git:/<span class="kw">/</span><span class="ot">github.com</span><span class="kw">/</span>hyphaltip/CSHL_NGS.git</code></pre>
<ol type="1">
<li><p>The data you need are available from here. There are directions at the bottom if you want to see how the data are obtained but not necessary for this tutorial.</p></li>
<li><p>We will use some data from a strain of yeast, W303 which has some drug resistance properties and mixed ancestry from the reference strain S288C. A project sequenced the genome and analyzed it to show amino acid alterations in ~800 genes. <a href="http://www.ncbi.nlm.nih.gov/pubmed/22977733">Ralser et al 2012</a>. We will attempt to replicate some aspects of this analysis (though note they did an assembly of the read not simply mapping them to the refeence).</p></li>
<li><p>These data are available in the data/example/W303_chrII_1.fastq.bz2 and W303_chrII_2.fastq.bz2 - you will need to uncompress with bunzip2.</p></li>
<li><p>The genome is available in data/genome/Saccharomyces.fa.gz - you need to uncompress this with gunzip data/genome/Saccharomyces.fa.gz</p></li>
</ol>
<h1 id="tutorial">Tutorial</h1>
<ol type="1">
<li><p>Trim FASTQ data for quality using <a href="https://github.com/najoshi/sickle">sickle</a> - run <code>sickle pe</code> to see how to run PE options</p></li>
<li><p>Compare the FASTQC quality report for one of the files (<em>1 or </em>2) files both before and after trimming. Set this up in the background so you can run it and do other things in the meantime.</p></li>
</ol>
<p><code>fastqc -h</code> to get help</p>
<ol start="3" type="1">
<li><p>Align reads to the genome using BWA. This requires you to also build and index for the genome. See the <a href="http://hyphaltip.github.com/CSHL_NGS/lecture/NGS_DNA.slides.html#slide34">lecture notes</a>. (bwa index)</p></li>
<li><p>Realign reads with Picard and GATK <a href="http://hyphaltip.github.com/CSHL_NGS/lecture/NGS_DNA.slides.html#slide41">based on lecture</a>. (picard)</p></li>
<li><p>Fix the Read groups see <a href="http://hyphaltip.github.com/CSHL_NGS/lecture/NGS_DNA.slides.html#slide41">this slide</a> (picard)</p></li>
<li><p>Call SNPs with SAMTools - refer to the SAMtools manpage on mpileup for more details. <a href="http://samtools.sourceforge.net/">http://samtools.sourceforge.net/</a> <a href="http://hyphaltip.github.com/CSHL_NGS/lecture/NGS_DNA.slides.html#slide43">this slide</a></p></li>
<li><p>Call SNPs with GATK, using <a href="http://hyphaltip.github.com/CSHL_NGS/lecture/NGS_DNA.slides.html#slide44">example from the lecture</a></p></li>
<li><p>Run Filtering steps on GATK output SNPs to remove potential biased or low-quality ones using options <a href="http://hyphaltip.github.com/CSHL_NGS/lecture/NGS_DNA.slides.html#slide47">provided in lecture</a>.</p></li>
<li><p>Calculate the total number of remaining SNPs. Count the lines or use <a href="http://vcftools.sourceforge.net/">vcftools</a>.</p></li>
<li><p>For advanced users, intersect this list of SNPs (in the VCF file) with the GFF for the genome to determine which SNPs are in coding regions. Read up on <a href="http://code.google.com/p/bedtools/">BEDTools</a>. The genome annotation in GFF is available in the folder where the genome was downloaded from <a href="http://yeastgenome.org">SGD</a>.</p></li>
<li><p>Open the genome file for Saccharomces in IGV. <a href="http://www.broadinstitute.org/igv/">http://www.broadinstitute.org/igv/</a> is here.</p></li>
</ol>
<p>Then add the GFF file as annotation track. Then BAM file, and VCF file in IGV to view the SNPs in context of the gene annotation and the read-depth</p>
<p>Feel free to try this also with your own favorite organism. Many datasets exist in the <a href="http://www.ncbi.nlm.nih.gov/sra">SRA</a> from genome resequencing. To extend the problem, download more than 4 strains so you can apply comparisons between individuals instead of just between one individual and the reference.</p>
<p>For example, here is the <a href="http://www.ncbi.nlm.nih.gov/bioproject/36679">Drosophila reference panel</a> which included sequencing 192 individuals. Or find something smaller (10 C.elegans for example).</p>
<h1 id="obtaining-the-datasets-from-the-public-archives-for-this-tutorial">Obtaining the datasets from the public archives for this tutorial</h1>
<ol start="2" type="1">
<li>Download the Saccharomyces genome from <a href="http://downloads.yeastgenome.org/sequence/S288C_reference/genome_releases/S288C_reference_genome_Current_Release.tgz">SGD genome release (2011)</a>. Uncompress this and get the .fsa file which is the genome.</li>
</ol>
<p>You could do this like</p>
<pre class="sourceCode perl"><code class="sourceCode perl">$ curl <span class="kw">-O</span>
http:/<span class="kw">/</span><span class="ot">downloads.yeastgenome.org</span><span class="kw">/s</span>equence/S288C_reference/genome_releases/S288C_reference_genome_Current_Release.tgz

$ tar zxf S288C_reference_genome_Current_Release.tgz</code></pre>
<p>Run this script to fix the chromosome names in the download file so the will match the GFF file.</p>
<pre class="sourceCode perl"><code class="sourceCode perl">perl CSHL_NGS/data/rename_seq.pl S288C_reference_genome_R64<span class="dv">-1-1_20110203</span>/S288C_reference_sequence_R64<span class="dv">-1-1_20110203</span>.fsa &gt; Saccharomyces.fa</code></pre>
<ul>
<li>You need to fix this GFF file so it doesn't have any sequence, to do this a grep to find where the '&gt;' lines are where the sequence as fasta is in there and find the first one.</li>
</ul>
<p>Commands to run</p>
<pre class="sourceCode perl"><code class="sourceCode perl"><span class="fu">grep</span> -n <span class="kw">&quot;</span><span class="st">&gt;</span><span class="kw">&quot;</span> S288C_reference_genome_R64<span class="dv">-1-1_20110203</span>/saccharomyces_cerevisiae_R64<span class="dv">-1-1_20110208</span>.gff
<span class="co"># note the number for the first &#39;&gt;&#39; - there are a few lines before that we want to drop so</span>
<span class="co"># and the last position in the file we want is 16425 so we can use head </span>
$ head -n <span class="dv">16425</span> S288C_reference_genome_R64<span class="dv">-1-1_20110203</span>/saccharomyces_cerevisiae_R64<span class="dv">-1-1_20110208</span>.gff &gt; saccharomyces_cerevisiae_R64<span class="dv">-1-1_20110208</span>.noseq.gff</code></pre>
<ul>
<li>Use this saccharomyces_cerevisiae_R64-1-1_20110208.noseq.gff for GFF file later needs.</li>
</ul>
<ol start="3" type="1">
<li><p>SRA data are available to download here. There is a script and table listing the sources for some strain data. You can use the Aspera pluging for fast downloading or wget/curl will work. The fastq-dump script as part of the sratoolkit is needed.</p>
<ul>
<li>The download script to obtain all the data is here <a href="https://github.com/hyphaltip/CSHL_NGS/blob/master/data/download.sh">https://github.com/hyphaltip/CSHL_NGS/blob/master/data/download.sh</a> or in the github repo you checked out - <code>CSHL_NGS/data/download.sh</code></li>
</ul></li>
</ol>
</body>
</html>
